core/tee/tee_svc_storage.c

```diff
@@ -444,18 +442,8 @@ TEE_Result syscall_storage_obj_del(unsigned long obj)
 	if (o->pobj == NULL || o->pobj->obj_id == NULL)
 		return TEE_ERROR_BAD_STATE;
 
-	if (IS_ENABLED(CFG_NXP_SE05X)) {
-		len = o->info.dataSize;
-		data = calloc(1, len);
-		if (!data)
-			return TEE_ERROR_OUT_OF_MEMORY;
-
-		res = o->pobj->fops->read(o->fh, o->info.dataPosition,
-					  data, &len);
-		if (res == TEE_SUCCESS)
-			crypto_storage_obj_del(data, len);
-		free(data);
-	}
+	/* The cryptographic layer might need to do house-keeping */
+	crypto_storage_obj_del(o);
```

Comment:
I prefer a configuration where the overhead can be reduced to zero when not needed. With this weak function always being called, we're still adding a few useless instructions for every platform. If this call was guarded with some IS_ENABLED() test, like before, it could be optimized away completely if not needed.